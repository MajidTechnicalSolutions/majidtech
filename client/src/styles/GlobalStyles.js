import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
body {
  margin: 0;
  font-family: "Lato";
  box-sizing: border-box;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

ul {
  list-style-type: none;
}

.active {
  color: ${(props) => props.theme.colors.orange}
}
`;

export default GlobalStyle;
