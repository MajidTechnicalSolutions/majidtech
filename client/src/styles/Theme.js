import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";

export const themes = {
  colors: {
    orange: "#dd6b2d",
    white: "#fff",
    black: "#121212",
    lightgrey: "#f3f3f4;",
    palegrey: "#FAFAFB",
    dimgrey: "#696969",
  },
  fontSize: {
    lg: "2rem",
    md: "1.5rem",
    sm: "1rem",
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={themes}>{children}</ThemeProvider>
);

export const shallowTheme = (tree) =>
  shallow(tree, {
    wrappingComponent: Theme,
  });

export const mountTheme = (tree) =>
  mount(tree, {
    wrappingComponent: Theme,
  });

export default Theme;
