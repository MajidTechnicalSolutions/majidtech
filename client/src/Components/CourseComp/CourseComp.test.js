import React from "react";
import CoursesComp from "./CourseComp";
import { shallow, mount } from "enzyme";
import { ThemeProvider } from "styled-components";
import { shallowTheme, mountTheme } from "../../styles/Theme";
import toJson from "enzyme-to-json";

const options = [{ key: 1, data: "Front-End" }];

const courses = [
  {
    image: 1,
    hours: 1,
    minutes: 30,
    id: 1,
    title: "sunt",
    disc:
      "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto",
  },
];

describe("mock test for CourseComp", () => {
  it("accepts courses props", () => {
    const wrapper = mountTheme(
      <CoursesComp courses={courses} options={options} />
    );
    expect(wrapper.props().courses).toEqual(courses);
    expect(wrapper.props().options).toEqual(options);
  });
  it("able to use props", () => {
    const wrapper = mountTheme(
      <CoursesComp courses={courses} options={options} />
    );

    const value = wrapper.find(".test").text();
    expect(value).toEqual("sunt");
  });
});

describe("snapshot test for CourseComp", () => {
  const tree = shallow(<CoursesComp courses={courses} options={options} />);
  expect(toJson(tree)).toMatchSnapshot();
});
