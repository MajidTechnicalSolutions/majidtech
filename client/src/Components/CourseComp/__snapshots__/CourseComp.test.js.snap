// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Fragment>
  <styled.h1>
    <styled.header>
      <styled.h1>
        Courses
      </styled.h1>
    </styled.header>
    <MenuBar
      options={
        Array [
          Object {
            "data": "Front-End",
            "key": 1,
          },
        ]
      }
    />
    <styled.div
      theme={
        Object {
          "colors": Object {
            "lightgrey": "#f3f3f4",
          },
        }
      }
    />
  </styled.h1>
  <styled.div
    theme={
      Object {
        "colors": Object {
          "lightgrey": "#FAFAFB",
        },
      }
    }
  >
    <styled.div>
      <styled.div
        theme={
          Object {
            "colors": Object {
              "lightgrey": "#f3f3f4",
            },
          }
        }
      >
        <styled.div
          alt="sunt"
          img={1}
        />
        <styled.div>
          <styled.div
            className="test"
          >
            sunt
          </styled.div>
          <Styled(styled.div)>
            quia et suscipit
suscipit recusandae consequuntur
            â€¦
          </Styled(styled.div)>
          <styled.div>
            <Styled(RiStackLine) />
            <Styled(RiTimerLine) />
            1
             
            30
          </styled.div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
